Move wheel handlers from window to canvas element in HTML

Schroedi opened the pull request:
Similar to https://github.com/godotengine/godot/pull/36557

At least in chrome, the following error is printed for each mouse wheel
rotation:
[Intervention] Unable to preventDefault inside passive event listener due to target being treated as passive. See https://www.chromestatus.com/features/6662647093133312

This PR moves the handler to the canvas and thereby fixes the error.

Tested on: Chrome and Firefox (MacOS), Firefox, Chrome(Android), Safari (IPad + MacOS)

*Bugsquad edit:* Fixes #31095.

Faless reviewed (changes requested):
First of all, sorry it took me so long to review this, I've been quite busy recently.

This is fine for `wheel`, but `mouseup` must be in window, so we can detect the up event when the mouse was pressed inside the canvas, and released outside the canvas.
Additionally, for consistency with other platforms, `mousemove` should be received when moved outside the canvas but original press was inside.

Schroedi committed:
Move mouse wheel handler from window to canvas element in HTML

Similar to https://github.com/godotengine/godot/pull/36557

At least in chrome, the following error is printed for each mouse wheel
rotation:
[Intervention] Unable to preventDefault inside passive event listener due to target being treated as passive. See https://www.chromestatus.com/features/6662647093133312

This PR moves the handler to the canvas and thereby fixes the error.

Tested on: Chrome and Firefox (MacOS), Firefox, Chrome(Android), Safari (IPad + MacOS)

Schroedi commented:
Good catch. I removed the move and button changes. 
If I remember correctly, it should be a change emscripten has to do. The wheel change is still valid, I think, to match the other platform's behaviors.

Faless reviewed (approved):
Looks good. Thanks!

akien-mga commented:
Cherry-picked for 3.2.2.
